graphql/admin/admin.go

```diff
@@ -326,8 +325,40 @@ var (
 		"deleteUser":  {resolve.IpWhitelistingMW4Mutation},
 		"deleteGroup": {resolve.IpWhitelistingMW4Mutation},
 	}
+	// mainHealth tracks the health of the main GraphQL server.
+	mainHealth = &GraphQLHealth{httpStatusCode: http.StatusServiceUnavailable, statusMsg: "init"}
 )
 
+// GraphQLHealth is used to report the health status of a GraphQL server.
+// It is required for kubernetes probing.
+type GraphQLHealth struct {
+	httpStatusCode int
+	statusMsg      string
+	// mux protects GraphQLHealth from simultaneous read/write, as an instance of this type could be
+	// used by multiple Go-routines
+	mux sync.RWMutex
+}
+
+func (g *GraphQLHealth) Status() (int, string) {
+	g.mux.RLock()
+	defer g.mux.RUnlock()
+	return g.httpStatusCode, g.statusMsg
+}
+
+func (g *GraphQLHealth) up() {
+	g.mux.Lock()
+	defer g.mux.Unlock()
+	g.httpStatusCode = http.StatusOK
```

Comment:
I feel like this is the wrong layer for this. The HTTP status code belongs in the http layer. Probably just store an enum or whatever at this layer